qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -251,8 +257,26 @@ setup_pools()
     rbd --cluster ${cluster} mirror pool enable ${POOL} pool
     rbd --cluster ${cluster} mirror pool enable ${PARENT_POOL} image
 
-    rbd --cluster ${cluster} mirror pool peer add ${POOL} ${remote_cluster}
-    rbd --cluster ${cluster} mirror pool peer add ${PARENT_POOL} ${remote_cluster}
+    if [ -z ${RBD_MIRROR_CONFIG_KEY} ]; then
+      rbd --cluster ${cluster} mirror pool peer add ${POOL} ${remote_cluster}
+      rbd --cluster ${cluster} mirror pool peer add ${PARENT_POOL} ${remote_cluster}
+    else
+      mon_map_file=${TEMPDIR}/${remote_cluster}.monmap
+      ceph --cluster ${remote_cluster} mon getmap > ${mon_map_file}
```

Comment:
@dillaman On teuthology [1] `TEMPDIR` at this point is empty by some reason:
```
2018-09-13T11:06:36.400 INFO:tasks.workunit.cluster1.client.mirror.smithi009.stderr:+ mon_map_file=/cluster2.monmap
2018-09-13T11:06:36.400 INFO:tasks.workunit.cluster1.client.mirror.smithi009.stderr:+ ceph --cluster cluster2 mon getmap
2018-09-13T11:06:36.401 INFO:tasks.workunit.cluster1.client.mirror.smithi009.stderr:/home/ubuntu/cephtest/clone.cluster1.client.mirror/qa/workunits/rbd/rbd_mirror_helpers.sh: line 265: /cluster2.monmap: Permission denied
```
This makes me crazy because I fails to understand why. And note, when cleanup is called `TEMPDIR` is ok again. And I can't reproduce this locally. I would thought it is some sort of a bug in /bin/sh, but I see this on different distributions, so it looks less likely.

[1] http://qa-proxy.ceph.com/teuthology/trociny-2018-09-13_06:37:31-rbd-wip-mgolub-testing-distro-basic-smithi/3015823/teuthology.log

Reply:
The original PR version used RBD_MIRROR_TEMDIR instead of TEMPDIR. I wonder if your test branch has the old version?